### PR TITLE
Require reason for Fail on Pass/Fail steps

### PR DESCRIPTION
**Pull request by Devorein**
### WHY

[Card](https://app.charmverse.io/charmverse/optional-require-reason-for-fail-on-pass-fail-steps-8571563609916741)

### CODE REVIEW

File: components/proposals/ProposalPage/components/ProposalEvaluations/components/Review/components/PassFailEvaluation.tsx [hunk 1]
```diff
@@ -134,12 +149,67 @@ export function PassFailEvaluation({
...
+            </Typography>
+          </FormLabel>
+          <Stack flexDirection='row' gap={1}>
+            {evaluation.failReasons.map((reason) => (
+              <Chip size='small' variant='outlined' key={reason} label={reason} sx={{ mr: 0.5 }} />
+            ))}
+          </Stack>
+        </Stack>
+      ) : null}
+      <ConfirmDeleteModal
```
Comment (mattcasey) [inline components/proposals/ProposalPage/components/ProposalEvaluations/components/Review/components/PassFailEvaluation.tsx:174]: this modal "ConfirmDeleteModal" should only be used for confirming delete. please use something else or rename it

File: components/settings/proposals/components/EvaluationDialog.tsx [hunk 2]
```diff
@@ -105,6 +119,77 @@ function StepActionButtonLabel({
...
+  setValue,
+  failReasons
+}: {
+  failReasons: string[];
+  setValue: UseFormSetValue<FormValues>;
+}) {
+  return (
+    <Box className='octo-propertyrow'>
+      <FieldLabel>Fail reasons</FieldLabel>
+      <Autocomplete<string, true, true, true>
```
Comment (mattcasey) [inline components/settings/proposals/components/EvaluationDialog.tsx:132]: this input should line up with the other inputs:
image

Also the UX is very vague. How do you add multiple ones? I think it woudl be better if we just had each reason on its own line, with a "Add reason" button below  the last input

File: components/proposals/ProposalPage/components/ProposalEvaluations/components/Review/components/PassFailEvaluation.tsx [hunk 3]
```diff
@@ -134,12 +149,67 @@ export function PassFailEvaluation({
...
+        onConfirm={async () => {
+          await onSubmitReview('fail');
+          setFailReasons([]);
+          failReasonModalPopupState.close();
+        }}
+        open={!!failReasonModalPopupState.isOpen}
+        question={
+          <Stack gap={1}>
+            <Typography>Please select at least one reason for declining this proposal.</Typography>
+            <Select
```
Comment (mattcasey) [inline components/proposals/ProposalPage/components/ProposalEvaluations/components/Review/components/PassFailEvaluation.tsx:188]: if i select a long option it gets cut off and is invisible:
image


The input should expand. i assumed we were only letting people select one option though

File: components/proposals/ProposalPage/components/ProposalEvaluations/components/Review/components/PassFailEvaluation.tsx [hunk 4]
```diff
@@ -134,12 +149,67 @@ export function PassFailEvaluation({
...
       </Card>
+      {evaluation.result === 'fail' && evaluation.failReasons.length ? (
+        <Stack mt={2} gap={1}>
+          <FormLabel>
+            <Typography component='span' variant='subtitle1'>
+              Fail reasons:
+            </Typography>
+          </FormLabel>
+          <Stack flexDirection='row' gap={1}>
+            {evaluation.failReasons.map((reason) => (
```
Comment (mattcasey) [inline components/proposals/ProposalPage/components/ProposalEvaluations/components/Review/components/PassFailEvaluation.tsx:168]: the chips should wrap and not be cut off:
image

File: components/proposals/ProposalPage/components/ProposalEvaluations/components/Review/components/PassFailEvaluation.tsx [hunk 5]
```diff
@@ -134,12 +149,67 @@ export function PassFailEvaluation({
...
+              value={failReasons}
+              onChange={(e) => {
+                setFailReasons(e.target.value as string[]);
+              }}
+              renderValue={(selected) =>
+                selected.map((reason) => (
+                  <Chip size='small' variant='outlined' key={reason} label={reason} sx={{ mr: 0.5 }} />
+                ))
+              }
+              multiple
```
Comment (mattcasey) [inline components/proposals/ProposalPage/components/ProposalEvaluations/components/Review/components/PassFailEvaluation.tsx:198]: users can only select only one reason at the moment

File: components/proposals/ProposalPage/components/ProposalEvaluations/components/Review/components/PassFailEvaluation.tsx (hunk 1 above)
Comment (mattcasey) [inline components/proposals/ProposalPage/components/ProposalEvaluations/components/Review/components/PassFailEvaluation.tsx:174]: you could probably just use our common Modal directly. theres nothing special about this one and its name is confusing here

File: components/settings/proposals/components/EvaluationContextMenu.tsx [hunk 6]
```diff
@@ -22,17 +22,17 @@ export function EvaluationContextMenu({ evaluation, onDelete, onDuplicate, onRen
     onDelete(evaluation.id);
   }
 
-  function renameEvaluation() {
-    onRename(evaluation);
+  function onEditEvaluation() {
+    onEdit(evaluation);
```
Comment (mattcasey) [inline components/settings/proposals/components/EvaluationContextMenu.tsx:26]: thx for cleaning up the method name :)

**Review by mattcasey — CHANGES_REQUESTED** [review]
I think it would be smart to wait and do the other story first? For pass/fail we need to start storing responses for multiple people and i think that the failReasons on the evaluation itself would immediately no longer be used

File: components/proposals/ProposalPage/components/ProposalEvaluations/components/Review/components/PassFailEvaluation.tsx (hunk 4 above)
Comment (Devorein) [reply components/proposals/ProposalPage/components/ProposalEvaluations/components/Review/components/PassFailEvaluation.tsx:168]: Fixed

File: components/proposals/ProposalPage/components/ProposalEvaluations/components/Review/components/PassFailEvaluation.tsx (hunk 1 above)
Comment (Devorein) [reply components/proposals/ProposalPage/components/ProposalEvaluations/components/Review/components/PassFailEvaluation.tsx:174]: Updated.

File: components/proposals/ProposalPage/components/ProposalEvaluations/components/Review/components/PassFailEvaluation.tsx (hunk 3 above)
Comment (Devorein) [reply components/proposals/ProposalPage/components/ProposalEvaluations/components/Review/components/PassFailEvaluation.tsx:188]: Nice catch. Updated.

File: components/proposals/ProposalPage/components/ProposalEvaluations/components/Review/components/PassFailEvaluation.tsx (hunk 5 above)
Comment (Devorein) [reply components/proposals/ProposalPage/components/ProposalEvaluations/components/Review/components/PassFailEvaluation.tsx:198]: Updated.

**Pull request closed without merging**